cloudshell/networking/cisco/autoload/cisco_port_attrs_service.py

```diff
@@ -26,8 +26,10 @@ def cisco_duplex_state_table(self):
             )
             if cisco_duplex_state_table:
                 self._cisco_duplex_state_table = {
-                    v.get("portIfIndex", "").lower(): k
-                    for k, v in cisco_duplex_state_table.iteritems()
+                    cisco_duplex_state_table.get(k, {})
```

Comment:
you'll always have the value for the key in a dictionary
just iterate over key, value in dict.items()